### PR TITLE
chore(test): move to jest.requireActual

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

I noticed that require.requireActual is deprecated, so that was changed here

**Result**

no more require.requireActual, and at the same time made castToJestMock consistent and slightly type-safer

### CODE REVIEW

File: test/utils/castToJestMock.ts [hunk 1]
```diff
@@ -1 +1,2 @@
-export const castToJestMock = (obj: any): jest.Mock => obj;
+export const castToJestMock = <T extends (...args: any[]) => any>(obj: T) =>
```
Comment (francoischalifour) [inline test/utils/castToJestMock.ts:1]: We need to fix this line in [accordance to our ESLint rule](https://app.circleci.com/pipelines/github/algolia/instantsearch.js/5623/workflows/77f5063e-54c3-4d4f-84f9-986ffcb8db4a/jobs/25854/parallel-runs/0/steps/0-106).

Comment (francoischalifour) [inline test/utils/castToJestMock.ts:1]: We need to fix this line in [accordance to our ESLint rule](https://app.circleci.com/pipelines/github/algolia/instantsearch.js/5623/workflows/77f5063e-54c3-4d4f-84f9-986ffcb8db4a/jobs/25854/parallel-runs/0/steps/0-106).

**Review by francoischalifour — APPROVED** [review]
Good to merge once the CI is fixed!

File: test/utils/castToJestMock.ts (hunk 1 above)
Comment (Haroenv) [reply test/utils/castToJestMock.ts:1]: ah yes of course! I didn't notice the squiggle

**Pull request merged**